addons/sale_purchase/models/sale_order_line.py

```diff
@@ -137,6 +138,39 @@ def _purchase_service_prepare_order_values(self, supplierinfo):
             'fiscal_position_id': fpos.id,
         }
 
+    def _purchase_service_get_price_unit_and_taxes(self, supplierinfo, purchase_order):
+        supplier_taxes = self.product_id.supplier_taxes_id.filtered(lambda t: t.company_id == purchase_order.company_id)
```

Comment:
As discuss in PM, but for history...

It should be more 'correct' since we apply taxes from the company of the PO company and not the current user that could differ.
In our case, it will allow as customization to assign specific company to some PO whatever the company of the current user / sale order.

Reply:
moved to second commit